core-bundle/src/EventListener/DataContainer/CteAllowedTypeListener.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\ContentModel;
+use Contao\CoreBundle\DependencyInjection\Attribute\AsCallback;
+use Contao\CoreBundle\Fragment\FragmentCompositor;
+use Contao\DataContainer;
+
+/**
+ * @internal
+ */
+class CteAllowedTypeListener
+{
+    public function __construct(
+        private readonly FragmentCompositor $compositor,
+    ) {
+    }
+
+    /**
+     * Switch to first allowed type if default type is not allowed (see #7100).
+     */
+    #[AsCallback(table: 'tl_content', target: 'config.oncreate')]
+    public function setAllowedType(string $strTable, int $insertID, array $values, DataContainer $dc): void
+    {
+        $objCte = ContentModel::findById($insertID);
```

Comment:
```suggestion
        $element = ContentModel::findById($insertID);
```

We only use the type prefix in variable names in legacy code.

Reply:
done.